src/garage/tf/policies/task_embedding_policy.py

```diff
@@ -0,0 +1,332 @@
+
+        """
+
+    def embedding_dist_info_sym(self,
+                                input_var,
+                                state_info_vars=None,
+                                name='embedding_dist_info_sym'):
+        """Return the symbolic distribution information about the embedding.
+
+        Args:
+            input_var(tf.Tensor): Symbolic variable for encoder input.
+            state_info_vars(dict): A dictionary whose values should contain
+                information about the state of the policy at the time it
+                receives the input.
+            name (str): Name for symbolic graph.
+
+        Returns:
+            dict[tf.Tensor]: Outputs of the symbolic graph of distribution
+                parameters.
+
+        """
+        return self.encoder.dist_info_sym(input_var, state_info_vars, name)
+
+    def embedding_dist_info(self, input_val, state_infos=None):
+        """Return the distribution information about the embedding.
+
+        Args:
+            input_val(tf.Tensor): Encoder input values.
+            state_infos(dict): A dictionary whose values should contain
+                information about the state of the policy at the time it
+                receives the input.
+
+        Returns:
+            dict[numpy.ndarray]: Distribution parameters.
+
+        """
+        return self.encoder.dist_info(input_val, state_infos)
+
+    @abc.abstractmethod
+    def dist_info_sym_under_task(self,
+                                 obs_var,
+                                 task_var,
+                                 state_info_vars=None,
+                                 name='default'):
+        """Build a symbolic graph of the action distribution given task.
+
+        Args:
+            obs_var (tf.Tensor): Symbolic observation input.
+            task_var (tf.Tensor): Symbolic task input.
+            state_info_vars (dict): Extra state information, e.g.
+                previous action.
+            name (str): Name for symbolic graph.
+
+        Returns:
+            dict[tf.Tensor]: Outputs of the symbolic graph of
+                action distribution parameters.
+
+        """
+
+    @abc.abstractmethod
+    def dist_info_sym_under_latent(self,
+                                   obs_var,
+                                   latent_var,
+                                   state_info_vars=None,
+                                   name='from_latent'):
+        """Build a symbolic graph of the action distribution given latent.
+
+        Args:
+            obs_var (tf.Tensor): Symbolic observation input.
+            latent_var (tf.Tensor): Symbolic latent input.
+            state_info_vars (dict): Extra state information, e.g.
+                previous action.
+            name (str): Name for symbolic graph.
+
+        Returns:
+            dict[tf.Tensor]: Outputs of the symbolic graph of distribution
+                parameters.
+
+        """
+
+    def get_action(self, observation):
```

Comment:
you can and should still provide this API. it assumes the observation is the onehot-augmented observation